src/rotator_library/client.py

```
@@ -1739,6 +1837,16 @@ async def _execute_provider_attempt(
                                 current_cred, classified_error, error_message
                             )
 
+                            # ponytail: mark credit-tier exhausted on quota; one_time permanent, daily resets via cron
+                            if classified_error.error_type == "quota_exceeded":
+                                try:
+                                    get_telemetry_manager().record_llm_call_outcome(
+                                        provider, current_cred or "", False,
+                                        error_type="quota_exceeded", model=model,
+                                    )
+                                except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)